src/core/logical_layers/rfc.md

```diff
@@ -0,0 +1,73 @@
+
+## Map Style Layers
+
+The [Style Specification](https://maplibre.org/maplibre-gl-js-docs/style-spec/) defines a map layer as a way of representing data.
+Such a layer is part of the map style and can be of type "fill", "line", "symbol", "circle" and so on.
+In the context of our application, we refer to such layers as map style layers, or just a style layer
+
+## Sources of layers
+
+If we divide layers by data delivery format, we can identify "vector", "raster" and "geojson" layers.
+However, this is a simplification - despite the fact that it directly affects the way the layer style is described, this division still refers to _layer sources_.
+At the same time, the vector data source type also has layers, which we call source-layer
+
+## Logical layers
+
+And, at the top of the list are the logical layers. A logical layer is a concatenation of several data and style layers linked together by a common theme.
+
+This is how these layers look like a tree:
+
+- logical layer
+  - style layers
+  - layer source
+    - source layer
+
+For the user of an application, the logical layer is the layer with which he interacts.
+For example, the logical layer "active contributors" includes:
+
+- a vector data source with multiple data layers (hexagon geometry h3, and coordinates with nicknames)
+- three style layers - text style, hexagon display style, glow style under the text.
+
+# Structure of a logical layer
+
+Each logical layer has additional data not related to the map, but necessary for other parts of the interface interacting with it.
+
+The logical layer interface can be found in the file [types/logicalLayers.ts](types/logicalLayer.ts).
+You can notice that we can divide them conditionally into two halves: data and states.
+The internal logic of the logical layer is described in [utils/logicalLayerFabric.ts](utils/logicalLayerFabric.ts)
+
+## Layers registry
+
+Depending on the application's task, you need to track changes in _all_ states of _one_ layer, or _specified_ state of _all_ layers. For example, an item in the Layer List tracks all changes in its layer, while a legend panel tracks changes in the legends of all layers.
+
+For this reason, states are stored in separate lists (layersMeta, layersLegends, layersMenus, enabledLayers, mountedLayers, etc.) from which each layer selects the state relevant to it.
+
+Separately stands layersRegistry - a list that contains all the logical layers, knows how to create and delete them.
+It is important to note that you do not need to add a logical layer to it, this should be done only if you want to allow the user to manage this layer in a standard way through the panel with the layers
+
+### isEnabled and isMounted
+
+The isEnabled and isMounted states deserves special mention. The layer gets `isEnabled` state if we want to show the layer to the user but for some reason cannot do it right now. Its main purpose is to _remember the choices made by the user_.
```

Comment:
```suggestion
The isEnabled and isMounted states deserves special mention. The layer gets `isEnabled` state if we want to show the layer to the user. Its main purpose is to _remember the choices made by the user_.
```
isEnabled stays on after layer being mounted, and this part implied that it turns off once layer was mounted